### PR TITLE
needs_configuration fix

### PR DESCRIPTION
**Pull request by zcsizmadia**
Fixes the issue that PVR addons must be configured to be able to enable them.

### CODE REVIEW

File: xbmc/pvr/addons/PVRClient.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ CPVRClient::CPVRClient(const cp_extension_t *ext) :
   m_strAvahiType = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@avahi_type");
   m_strAvahiIpSetting = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@avahi_ip_setting");
   m_strAvahiPortSetting = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@avahi_port_setting");
-  m_bNeedsConfiguration = !(CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@needs_configuration") == "false");
+  m_bNeedsConfiguration = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@needs_configuration") == "true";
```
Comment (FernetMenta) [inline xbmc/pvr/addons/PVRClient.cpp]: could you please explain why this makes a difference?

**Comment by zcsizmadia** [comment]
If you have a PVR addon with settings, the addon must be configured before it can be enabled. Even if the addon would work with the built-in default values. Right now Kodi does not even notifies the user on the GUI or in the log file that the addon must be configured.

My pvr.hdhomerun addon (https://github.com/kodi-pvr/pvr.hdhomerun) would work fine without configuring any settings, since it detects everything needed automatically. But because it is not configured the addon manager disables it  right after I enable it (unless a click on configure manually which will create the user setting xml file).

My workaround now is that I have "need_configuration"="false" in my addon.xml. But I believe the default should be the opposite.

If a PVR addon needs configuration manually from the user, the addon.xml should contain  "need_configuration"="true", otherwise the addon uses its own default settings.

**Comment by FernetMenta** [comment]
my point is that you changed something like:

result = !(exp == false)

to

results = (exp == true)

**Comment by FernetMenta** [comment]
if this changes anything, there is a bug somewhere that needs to get fixed

**Comment by stefansaraev** [comment]
+1. finaly someone else who thinks that current pvrmanager behaviour is meh.

**Comment by zcsizmadia** [comment]
FernetMenta:
It is more like result = !(exp == "false"), where exp is "" if property is not specified.

The issue is that the GetExp function returns "" if "needs_configuration" is not specified which will result in m_bNeedsConfiguration being true, which I believe should be the opposite.

**Comment by FernetMenta** [comment]
imo this is works as desired. if an addon does not explicitly announce that it does not require configuration, it requires config. the addon itself can control this.

**Comment by zcsizmadia** [comment]
My interpretation was needs_configuration should be specified if it needs configuration, like the property name suggests and not the other way around. 

Since the PVR addons provide default configuration values even if the addon is not configured manually.

So this patch is more like fixing what should be the default value for the m_bNeedsConfiguration field when "needs_configuration" is not specified, meaning addon does not need configuration.

**Comment by MartijnKaijser** [comment]
I say make it mandatory that pvr add-ons actually set that value

**Comment by zcsizmadia** [comment]
Most of the PVR addons must be configured since you have to set the server address, port, username, password, XML input file etc.

In my case the addon discovers everything, but the addon have some tweaks the user can modify IF he really wants to.

Other option is that if the addon needs configuration at least there is an entry in the debug log or shows a notification on the GUI.

This small thing caused me some major headache, why Kodi disables my addon right after I enabled it :)

**Comment by FernetMenta** [comment]
so this is more a API doc and developer issue than something we should convey to the user. as you mentioned most addons require config and are happy with the current state. if you set the property correctly on your addon, you and your users are happy too

**Comment by zcsizmadia** [comment]
FernetMenta:
I still belive it is kind of confusing leaving the current implementation. I've tried enabling the Myth TV PVR addon. If I just enable it, it gets disabled immediately. If needs_configuration false would be the default, the addon would be enabled and start complaining that cannot connect to server, making it trivial that must be configured. (I've checked this by adding "needs_configuration"="false" to its addon.xml.

Of course I think you are right when you say it sounds like be an API documentation issue (as well).

I will  add a patch which create a LOG_DEBUG entry if the addon is disabled, because it is not configured. At least someone else might avoid this issue.

**Comment by hudokkow** [comment]
I agree with @MartijnKaijser 
> I say make it mandatory that pvr add-ons actually set that value

What's the preferred/proper solution? Change core or change addon.xml.in for every addon?

**Comment by MartijnKaijser** [comment]
Make sure that every addon adds the true/false in addon.xml and let core open up the settings dialog if it is set to true on first enable of that addon.

**Comment by FernetMenta** [comment]
> Make sure that every addon adds the true/false in addon.xml

how would you do this? if it is not enforced by the code, a dev my find himself in the same situation as @zcsizmadia was in

**Comment by zcsizmadia** [comment]
There are very limited number of PVR addons so far, so changing their addon.xml.in is doable. If we have the "luxury" to "force" the addon.xml.in changes, I would not change how the default "needs_configuration" is defined by the current implementation. 

The additional notification/dialog would be nice to signal user, or just simply disable the enable button, until it is configured (if it is needed).

**Comment by stefansaraev** [comment]
this is not needed anymore. pvrmanager does not own addons anymore and will not try disabling not configured clients. now it is dead code (I'll send cleanup PR later)

**Pull request closed without merging**